src/tests/nativeaot/CustomMain/CustomMain.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+class Program
```

Comment:
Could you also add a [module initializer](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/module-initializers) that increments some static int variable and then in Main check that the static variable is 1? (To make sure it executed, and it executed once)

Reply:
I added a module initializer that increments a static variable. Along with the class constructor, this should ensure that everything is executed once.